src/Core/Domain/Order/Command/AddOrderFromBackOfficeCommand.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Order\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Cart\ValueObject\CartId;
+use PrestaShop\PrestaShop\Core\Domain\Employee\ValueObject\EmployeeId;
+
+/**
+ * Adds new order from given cart.
+ */
+class AddOrderFromBackOfficeCommand
```

Comment:
@matks I've introduced specific command for adding order from BO, wdyt? You won't be able to use it in FO.

Reply:
I dont know 🤔 I feel weird about it. Is it really "Add order from BO" usecase ?

For example you can register a customer from many places: using front-office register form, imported in a CSV file, created from the BO add page...
I would rather call these `CustomerSelfRegisterCommand` for FO, `CreateCustomerCommand` for BO and `ImportCustomerCommand` for import 😉 .

Also dont forget the API can perform read/write as complete as the BO. So if we reuse Commands in the API, we're going to have a AddOrderFromBOCommand used in the webservice. Doesn't it feel weird 😉 ?

Reply:
> Is it really "Add order from BO" usecase ?

It is, see https://prnt.sc/nip9br, it's is added from BO.

> Doesn't it feel weird ?

Well, it kind of does feel weird, but adding order from BO is a bit specific. You need an employee in context, you can use `BoOrder` as payment method and some other minor differences compared to placing order in FO.

It could maybe be renamed to `AddOrderByEmployeeCommand`, wdyt?

P.S. For me, it seems that `PlaceOrder` would be more correct description of this action, but Add/Create is used in BO. :/

Reply:
If you can use `BoOrder` then `AddOrderFromBackOfficeCommand` makes sense 👍